src/Container/Container.php

```diff
@@ -0,0 +1,1173 @@
+		);
+	}
+
+	/**
+	 * Get all callbacks for a given type.
+	 *
+	 * @param string $abstract
+	 * @param object $object
+	 * @param array  $callbacksPerType
+	 *
+	 * @return array
+	 */
+	protected function getCallbacksForType( $abstract, $object, array $callbacksPerType ) {
+		$results = [];
+
+		foreach ( $callbacksPerType as $type => $callbacks ) {
+			if ( $type === $abstract || $object instanceof $type ) {
+				array_push( $results, ...$callbacks );
+			}
+		}
+
+		return $results;
+	}
+
+	/**
+	 * Fire an array of callbacks with an object.
+	 *
+	 * @param mixed $object
+	 * @param array $callbacks
+	 *
+	 * @return void
+	 */
+	protected function fireCallbackArray( $object, array $callbacks ) {
+		foreach ( $callbacks as $callback ) {
+			$callback( $object, $this );
+		}
+	}
+
+	/**
+	 * Get the container's bindings.
+	 *
+	 * @return array
+	 */
+	public function getBindings() {
+		return $this->bindings;
+	}
+
+	/**
+	 * Get the alias for an abstract if available.
+	 *
+	 * @param string $abstract
+	 *
+	 * @return string
+	 */
+	public function getAlias( $abstract ) {
+		if ( ! isset( $this->aliases[ $abstract ] ) ) {
+			return $abstract;
+		}
+
+		return $this->getAlias( $this->aliases[ $abstract ] );
+	}
+
+	/**
+	 * Get the extender callbacks for a given type.
+	 *
+	 * @param string $abstract
+	 *
+	 * @return array
+	 */
+	protected function getExtenders( $abstract ) {
+		$abstract = $this->getAlias( $abstract );
+
+		return isset( $this->extenders[ $abstract ] ) ? $this->extenders[ $abstract ] : [];
+	}
+
+	/**
+	 * Remove all of the extender callbacks for a given type.
+	 *
+	 * @param string $abstract
+	 *
+	 * @return void
+	 */
+	public function forgetExtenders( $abstract ) {
+		unset( $this->extenders[ $this->getAlias( $abstract ) ] );
+	}
+
+	/**
+	 * Drop all of the stale instances and aliases.
+	 *
+	 * @param string $abstract
+	 *
+	 * @return void
+	 */
+	protected function dropStaleInstances( $abstract ) {
+		unset( $this->instances[ $abstract ], $this->aliases[ $abstract ] );
+	}
+
+	/**
+	 * Remove a resolved instance from the instance cache.
+	 *
+	 * @param string $abstract
+	 *
+	 * @return void
+	 */
+	public function forgetInstance( $abstract ) {
+		unset( $this->instances[ $abstract ] );
+	}
+
+	/**
+	 * Clear all of the instances from the container.
+	 *
+	 * @return void
+	 */
+	public function forgetInstances() {
+		$this->instances = [];
+	}
+
+	/**
+	 * Flush the container of all bindings and resolved instances.
+	 *
+	 * @return void
+	 */
+	public function flush() {
+		$this->aliases         = [];
+		$this->resolved        = [];
+		$this->bindings        = [];
+		$this->instances       = [];
+		$this->abstractAliases = [];
+	}
+
+	/**
+	 * Get the globally available instance of the container.
+	 *
+	 * @return static
+	 */
+	public static function getInstance() {
+		if ( is_null( static::$instance ) ) {
+			static::$instance = new static;
```

Comment:
@JasonTheAdams did you plan to declare this property dynamically because I did not find this property in class properties